### PR TITLE
Use pathname + querystring for data loading

### LINKED ISSUE #262: useDeno hook doesn't track querystring parameter dependencies
I was using data from the query parameters to fetch data inside of a useDeno hook... but it seems there's no dependency tracking for query params.  

Looking here it uses only the pathname:
https://github.com/alephjs/aleph.js/blob/master/framework/react/hooks.ts#L40

I think it would make sense to use the full URL instead.
(the use case is something like search + pagination /search?q=hello&from=10&limit=20)

### PR DESCRIPTION
**Pull request by stereosteve**
Adds `getFullPath` helper for combining pathname + querystring, which should be used as key for all data fetching.

When using in data fetching xhr, it can be URL encoded:
`/_aleph/data/{encodeURIComponent(fullPath)}.json`

Also fixes bug in Anchor.isCurrent being incorrect when adding search:
/foo -> /foo?bar=1

Closes #262

I tested with `examples/anchor-links` and confirmed fixed all the Query links.  
I also did a `build` and the `dist/_aleph/data` output looked okay.  Basic SSG + data seemed to work, but I didn't do much before + after testing with SSG + data use case, so let me know if it might break something there.

### CODE REVIEW

File: framework/core/routing.ts [hunk 1]
```diff
@@ -129,6 +129,7 @@ export class Routing {
   createRouter(location?: { pathname: string, search?: string }): [RouterURL, RouteModule[]] {
     const loc = location || (window as any).location || { pathname: '/' }
     const url = rewriteURL(loc.pathname + (loc.search || ''), this._basePath, this._rewrites)
+    const slug = encodeURIComponent(loc.pathname + (loc.search || ''))
```
Comment (stereosteve) [inline framework/core/routing.ts:132]: maybe this should use output of `rewriteURL` (`url` instead of `loc`)?

**Comment by ije** [comment]
@stereosteve thanks,looks greate! i prefer use `toString` (function) instead of  `slug`, that is more common.

**Comment by stereosteve** [comment]
Okay... since RouterURL isn't a class I can add a separate helper function like `const pathString = getPathString(url)` instead of adding the `slug` field.  Would that work?

**Comment by ije** [comment]
@stereosteve i'm thinking we should encode the `fullpath` to base64 string that can accept any `search` content, since we need to save the ssr data on disk at SSR phase.

**Comment by stereosteve** [comment]
@ije yeah that makes sense... at the moment the code in `app.ts` that writes data to `_aleph/data` directory only uses `pathname`... but eventually when there is support for `getStaticPaths` it will make sense to base64 encode.

I'll make that change.

**Comment by stereosteve** [comment]
@ije updated to use base64 encoding instead of URL encoding.  Rebased off master.  Tests work locally, but fail in CI.  You know why?

lmk if there's anything else, or if you want me to squash to 1 commit.

**Comment by stereosteve** [comment]
Trying to test out the "build" on this branch:

```
ALEPH_DEV=true deno run -A --unstable --import-map=./import_map.json cli.ts build ./examples/anchor-links -L debug
```

I get:

```
DEBUG compile 'http://localhost:2020/framework/react/components/Anchor.ts' in 10ms
INFO   {} polyfills.js (ES2015) • 207KB
 > examples/anchor-links/.aleph/production/-/cdn.esm.sh/v41/deno_std_node_fs.js:54:4: error: Transforming async generator functions to the configured target environment ("es2015") is not supported yet
    54 │     async *iterate() {
       ╵     ~~~~~

 > examples/anchor-links/.aleph/production/-/cdn.esm.sh/v41/deno_std_node_fs.js:5291:39: error: Big integer literals are not available in the configured target environment ("es2015")
    5291 │         size: toBigInt(origin.size) || 0n,
         ╵                                        ~~

 > examples/anchor-links/.aleph/production/-/cdn.esm.sh/v41/deno_std_node_fs.js:5300:65: error: Big integer literals are not available in the configured target environment ("es2015")
    5300 │         mtimeNs: origin.mtime ? BigInt(origin.mtime.getTime()) * 1000000n : null,
         ╵                                                                  ~~~~~~~~

 > examples/anchor-links/.aleph/production/-/cdn.esm.sh/v41/deno_std_node_fs.js:5301:65: error: Big integer literals are not available in the configured target environment ("es2015")
    5301 │         atimeNs: origin.atime ? BigInt(origin.atime.getTime()) * 1000000n : null,
         ╵                                                                  ~~~~~~~~

 > examples/anchor-links/.aleph/production/-/cdn.esm.sh/v41/deno_std_node_fs.js:5302:77: error: Big integer literals are not available in the configured target environment ("es2015")
    5302 │         birthtimeNs: origin.birthtime ? BigInt(origin.birthtime.getTime()) * 1000000n : null,
         ╵                                                                              ~~~~~~~~

 > examples/anchor-links/.aleph/production/-/cdn.esm.sh/v41/deno_std_node_fs.js:5319:65: error: Big integer literals are not available in the configured target environment ("es2015")
    5319 │         ctimeNs: origin.mtime ? BigInt(origin.mtime.getTime()) * 1000000n : null
         ╵                                                                  ~~~~~~~~

error: Uncaught (in promise) Error: Build failed with 6 errors:
examples/anchor-links/.aleph/production/-/cdn.esm.sh/v41/deno_std_node_fs.js:54:4: error: Transforming async generator functions to the configured target environment ("es2015") is not supported yet
examples/anchor-links/.aleph/production/-/cdn.esm.sh/v41/deno_std_node_fs.js:5291:39: error: Big integer literals are not available in the configured target environment ("es2015")
examples/anchor-links/.aleph/production/-/cdn.esm.sh/v41/deno_std_node_fs.js:5300:65: error: Big integer literals are not available in the configured target environment ("es2015")
examples/anchor-links/.aleph/production/-/cdn.esm.sh/v41/deno_std_node_fs.js:5301:65: error: Big integer literals are not available in the configured target environment ("es2015")
examples/anchor-links/.aleph/production/-/cdn.esm.sh/v41/deno_std_node_fs.js:5302:77: error: Big integer literals are not available in the configured target environment ("es2015")
```

Works okay on `master`...  🤔

**Comment by ije** [comment]
@stereosteve currently the CSS testing with esbuild is not stable, just ignore it, i will fix it later

**Pull request merged**